src/ui/wxWidgets/SafeCombinationEntryDlg.cpp

```diff
@@ -366,13 +346,19 @@ wxIcon SafeCombinationEntryDlg::GetIconResource( const wxString& WXUNUSED(name)
 void SafeCombinationEntryDlg::OnOk( wxCommandEvent& )
 {
   // For the validation process, put the full file path name back into the combo box.
+  // Calling 'EllipsizeFilePathname' will undo this.
   m_filenameCB->ChangeValue(m_filename);
 
-  if (Validate() && TransferDataFromWindow()) {
```

Comment:
If we keep the logic this way, we can reduce the number of places EllipsizeFilePathname is called, i.e.
```
if (Validate() && TransferDataFromWindow()) {
...
}
EllipsizeFilePathname();
```